packages/mui-joy/src/Radio/RadioProps.ts

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { OverridableStringUnion, OverrideProps } from '@mui/types';
+import { UseSwitchProps } from '@mui/base/SwitchUnstyled';
+import { SxProps } from '../styles/defaultTheme';
+import { ColorPaletteProp, VariantProp } from '../styles/types';
+
+export type RadioSlot = 'root' | 'radio' | 'action' | 'input' | 'label';
+
+export interface RadioPropsVariantOverrides {}
+
+export interface RadioPropsColorOverrides {}
+
+export interface RadioPropsSizeOverrides {}
+
+export interface RadioTypeMap<P = {}, D extends React.ElementType = 'span'> {
+  props: P &
+    UseSwitchProps & {
+      /**
+       * The icon to display when the component is checked.
+       */
+      checkedIcon?: React.ReactNode;
+      /**
+       * Class name applied to the root element.
+       */
+      className?: string;
+      /**
+       * The component used for the Root slot.
+       * Either a string to use a HTML element or a component.
+       */
+      component?: React.ElementType;
+      /**
+       * The props used for each slot inside the Input.
+       * @default {}
+       */
+      componentsProps?: {
+        root?: React.ComponentPropsWithRef<'span'>;
+        radio?: React.ComponentPropsWithRef<'span'>;
+        action?: React.ComponentPropsWithRef<'span'>;
+        input?: React.ComponentPropsWithRef<'input'>;
+        label?: React.ComponentPropsWithRef<'label'>;
+      };
+      /**
+       * The color of the component. It supports those theme colors that make sense for this component.
+       * @default 'neutral'
+       */
+      color?: OverridableStringUnion<
+        Exclude<ColorPaletteProp, 'context'>,
+        RadioPropsColorOverrides
+      >;
+      /**
+       * If `true`, the checked icon is removed and the selected variant is applied on the `action` element instead.
+       * @default false
+       */
+      disableIcon?: boolean;
+      /**
+       * The label element at the end the radio.
+       */
+      label?: React.ReactNode;
+      /**
+       * The `name` attribute of the input.
+       */
+      name?: string;
+      /**
+       * If `true`, the root element's position is set to initial which allows the action area to fill the nearest positioned parent.
+       * This prop is useful for composing Radio with ListItem component.
+       * @default false;
+       */
+      overlay?: boolean;
```

Comment:
Really enjoyed this prop, it's very powerful! Though I wonder if we couldn't call it something more clear at a first glance. @samuelsycamore for input.

Reply:
I'm not sure I understand exactly what it would be used for. Can you elaborate on what it does?

Reply:
By default, the radio looks like this

<img width="127" alt="Screen Shot 2565-04-20 at 09 53 06" src="https://user-images.githubusercontent.com/18292247/164140930-317ed2be-e675-48bd-afe2-a3fbb07d5b85.png">

The intention of this prop is to make the developer experience easier if you want to forward the action area and the focus to the upper parent.

<img width="409" alt="Screen Shot 2565-04-20 at 09 53 14" src="https://user-images.githubusercontent.com/18292247/164141086-aeb29ffd-e8db-4f7c-9ba9-4af570ad05d8.png">

This prop lets you create complex radio UI like in the Apple store without having to create a custom radio (It takes care of the CSS for you).

<img width="404" alt="Screen Shot 2565-04-20 at 09 57 37" src="https://user-images.githubusercontent.com/18292247/164141298-cff8ac83-89a1-4173-847d-8b11b67cc8a6.png">

Reply:
Ah I see, that makes sense when you explain it visually! Very cool feature, and very tough to describe as a prop name. "Overlay" seems to imply that it's a layer that sits on top of the component, but what it's really doing is something a bit different—moving the focus of the radio button to its container. I'll sleep on it. 😅

Reply:
> Ah I see, that makes sense when you explain it visually! Very cool feature, and very tough to describe as a prop name. "Overlay" seems to imply that it's a layer that sits on top of the component, but what it's really doing is something a bit different—moving the focus of the radio button to its container. I'll sleep on it. 😅

I am happy to change the name! There are other components that have this feature as well. (eg. Link, Checkbox)

Reply:
Anyway, I don't think it will block the PR right? @danilo-leal

Reply:
Yeah, I think we could keep this one in mind and later on change it. Other than this small naming aspect, the whole component is looking great!